boards/wio_terminal/src/storage.rs

```diff
@@ -42,3 +47,62 @@ pub struct SDCard {
     /// SD Card detect pin
     pub det: Pd21<Input<Floating>>,
 }
+
+/// Type alias for SERCOM6 initialized for interacting with an SD card.
+pub type SDCardSPI =
+    SPIMaster6<Sercom6Pad2<Pc18<PfC>>, Sercom6Pad0<Pc16<PfC>>, Sercom6Pad1<Pc17<PfC>>>;
+
+/// Type alias for the SDCard chip select pin.
+pub type SDCardCS = Pc19<Output<PushPull>>;
+
+impl SDCard {
+    /// Initialize the SDCard and its corresponding SPI bus peripheral.
+    pub fn init(
+        self,
+        clocks: &mut GenericClockController,
+        sercom6: SERCOM6,
+        mclk: &mut MCLK,
+        port: &mut Port,
+    ) -> Result<(SDCardSPI, SDCardCS, Pd21<Input<Floating>>), ()> {
+        let gclk0 = clocks.gclk0();
+        let spi = SPIMaster6::new(
+            &clocks.sercom6_core(&gclk0).ok_or(())?,
+            3.mhz(),
+            spi::Mode {
+                phase: spi::Phase::CaptureOnFirstTransition,
+                polarity: spi::Polarity::IdleLow,
+            },
+            sercom6,
+            mclk,
+            (
+                self.miso.into_pad(port),
+                self.mosi.into_pad(port),
+                self.sck.into_pad(port),
+            ),
+        );
+
+        // Configure the chip select pin as a push-pull output.
+        let cs = self.cs.into_push_pull_output(port);
+        // let dc = self.dc.into_push_pull_output(port);
+        // let reset = self.reset.into_push_pull_output(port);
+        //
+        // // Create a SPIInterface over the peripheral, then create the ILI9341 driver
+        // // using said interface and set its default orientation.
+        // let interface = SPIInterface::new(spi, dc, cs);
+        // let mut ili9341 = Ili9341::new(interface, reset, delay).map_err(|_| ())?;
+        // ili9341
+        //     .set_orientation(Orientation::LandscapeFlipped)
+        //     .map_err(|_| ())?;
+        //
+        // // Configure the backlight pin as a push-pull output; unfortunately this pin
+        // // does not appear to support PWM.
+        // //   HIGH - backlight enabled
+        // //   LOW  - backlight disabled
+        // let mut backlight = self.backlight.into_push_pull_output(port);
+        // backlight.set_high()?;
+
+        // Return a result consisting of a Tuple containing the display driver and
+        // backlight pin.
```

Comment:
```suggestion
```
I don't think this needs to be included 😉

Reply:
Oooof good catch!